src/util.rs

```diff
@@ -3,15 +3,55 @@ use bitcoin::hashes::Hash;
 use bitcoin::{PubkeyHash, Script};
 
 use crate::miniscript::context;
+use crate::miniscript::satisfy::{PartialSatisfaction, Placeholder};
 use crate::prelude::*;
-use crate::{ScriptContext, ToPublicKey};
+use crate::{MiniscriptKey, ScriptContext, ToPublicKey};
 pub(crate) fn varint_len(n: usize) -> usize {
     bitcoin::VarInt(n as u64).len()
 }
 
+pub(crate) trait ItemSize {
+    fn size(&self) -> usize;
+}
+
+impl<Pk: MiniscriptKey> ItemSize for Placeholder<Pk> {
+    fn size(&self) -> usize {
+        match self {
+            Placeholder::PubkeyHash(_, size) => *size,
+            Placeholder::Pubkey(_, size) => *size,
+            Placeholder::EcdsaSigPk(_) | Placeholder::EcdsaSigHash(_) => 73,
+            Placeholder::SchnorrSig(_, _) | Placeholder::SchnorrSigHash(_, _) => 66,
```

Comment:
Yes, I think it can be done. We'll then store the sighash in the `RequiredSig` enum so that you know how to produce the signature so that it matches the expected size.

Just to clarify: we use `66` here because it counts the `OP_PUSH` byte as well, but with sighash_default it can drop to `65`.

Reply:
Hmm why include the varint for the size here (I assume that's what you mean by `OP_PUSH`)? the element size itself seems more useful because you can easily go from `elem_size -> varint_len + elem_size` but the other way around is harder (impossible?). Also it doesn't look like for `TapControlBlock` and `TapScript` you are including the varint_len.